text-motions.c

```diff
@@ -586,25 +586,37 @@ size_t text_function_end_prev(Text *txt, size_t pos) {
 	return text_function_end_direction(txt, pos, -1);
 }
 
-size_t text_block_start(Text *txt, size_t pos) {
+static size_t text_block_direction(Text *txt, size_t pos, int direction) {
 	int nest = 0;
-	char c;
+	char c, target, nest_target;
 	bool instring = false;
 	Iterator it = text_iterator_get(txt, pos);
-	while (text_iterator_byte_prev(&it, &c)) {
+
+	target = (direction > 0 ? '}' : '{');
+	nest_target = (direction > 0 ? '{' : '}');
```

Comment:
Additionally, I find even when writing plain text, being able to jump to the start/end of a `()` pair is pretty handy and I don't have to context switch, or put the editor into a C-language mode.

(although admittedly you could turn on the `(` block matching for all syntaxes)